notebook/before-notebook.d/10_workdir.sh

```diff
@@ -28,7 +29,18 @@ git_fetch() {
   local checkout="$2"
 
   if [[ ! -d "$checkout/.git" ]]; then
-    git clone "$remote" $checkout
+    # Splits the remote into remote@ref. Note we must always ensure to include
+    # an @ here, or the checkout will fail
+    remote_url=${remote%@*}
+    reference=${remote#*@}
+    git clone "$remote_url" $gittempdir
+    pushd $gittempdir
+    git checkout "$reference"
+    popd
+    # Since $workdir is not empty, files need to be copied into it
+    rm -rf $gittempdir/.git
+    cp -r $gittempdir/. $checkout
+    rm -rf $gittempdir
```

Comment:
Something like this should also work

```shell
git clone "$remote_url" "$gittempdir"
pushd "$gittempdir"
git archive --format=tar "$reference" | tar xf - -C "$checkout"
popd
rm -rf "$gittempdir"
```

Only saves you 2 lines though! But I think the archive will also mean you won't accidentally copy .gitmodules or other files like .gitignore and things.